drivers/wifi/nrf700x/src/fw_load.c

```diff
@@ -22,7 +17,16 @@
 #endif /* CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_XIP */
 
 #include <zephyr/logging/log.h>
+#if defined(CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_XIP) && defined(CONFIG_NORDIC_QSPI_NOR)
+/* For NRF QSPI NOR special handling is needed for this file as all RODATA of
+ * this file is stored in external flash, so, any use of RODATA has to be protected
+ * by disabling XIP and enabling it again after use. This means no LOG_* macros
+ * (buffered) or buffered printk can be used in this file, else it will crash.
+ */
```

Comment:
But how can this work then, if you compile this in one version of ncs and with one toolchain, you will get one address, if you build it in another, you will get a completely different address, so if someone updates their app then this is going to presumably fault?

Reply:
I don't understand. The base address can change but the content always points to the patch, so, no issues.

Reply:
there is a function here, `nrf_wifi_fw_load()`, so if this is in QSPI, how can you guarantee that this will never change address or definition? And what happens if as mentioned in another part, if the patch is part the way through an update when the device reboots and the code is incomplete, behaviour is completely undefined and device is bricked because it'll probably bootloop?

Reply:
We are only relocating `RODATA` not `text.

>  if the patch is part the way through an update when the device reboots and the code is incomplete, behaviour is completely undefined and device is bricked because it'll probably bootloop?

On reboot patch will be read again and downloaded.

Reply:
Discussed offline: DFU needs a separate discussion on how to do this patch partition upgrade for both XIP/non-XIP and using two partitions to avoid bricking Wi-Fi, this will be taken separately.